lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1133,13 +1133,23 @@ class IOSTextEditingStrategy extends GloballyPositionedTextEditingStrategy {
 
     _addTapListener();
 
-    // On iOS, blur is trigerred if the virtual keyboard is closed or the
-    // browser is sent to background or the browser tab is changed.
+    // On iOS, blur is trigerred in the following cases:
     //
-    // Since in all these cases, the connection needs to be closed,
-    // [domRenderer.windowHasFocus] is not checked in [IOSTextEditingStrategy].
+    // 1. The browser app is sent to the background (or the tab is changed). In
+    //    this case, the window loses focus (see [domRenderer.windowHasFocus]),
+    //    so we close the input connection with the framework.
+    // 2. The user taps on another focusable element. In this case, we refocus
+    //    the input field and wait for the framework to manage the focus change.
+    // 3. The virtual keyboard is closed by tapping "done". We can't detect this
+    //    programmatically, so we end up refocusing the input field. This is
+    //    okay because the virtual keyboard will hide, and as soon as the user
+    //    taps the text field again, the virtual keyboard will up.
```

Comment:
nit: "will _come_ up"?

Reply:
Nice catch! I meant to say "show up" but "come up" is also good :)